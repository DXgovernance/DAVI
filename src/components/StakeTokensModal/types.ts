import { BigNumber, providers } from 'ethers';
import { ERC20Info } from 'hooks/Guilds/erc20/useERC20Info';
import { ERC20, ERC20Guild } from 'types/contracts';
import { GuildConfigProps } from 'hooks/Guilds/guild/useGuildConfig';
import React from 'react';
export interface StakeTokensModalProps {
  isOpen: boolean;
  onDismiss: () => void;
  token?: ERC20Info;
  children?: React.ReactElement;
}

export interface StakeTokensFormsProps {
  token: Token;
  userVotingPower: BigNumber;
  guild: Guild;
  createTransaction: (
    summary: string,
    txFunction: () => Promise<providers.TransactionResponse>
  ) => void;
  isRepGuild: boolean;
}
interface Guild {
  contract: ERC20Guild;
  config: GuildConfigProps;
  totalLocked: BigNumber;
}
export interface Token {
  name: string;
  allowance: BigNumber;
  balance: BigNumber;
  info: ERC20Info;
  contract: ERC20;
}

export interface BalanceWidgetWrapperProps {
  token: Token;
  stakeAmount: BigNumber;
  setStakeAmount: (amount: BigNumber) => void;
  roundedBalance: number;
}

export interface StakeTokenButtonProps {
  isRepGuild: boolean;
  stakeAmount: BigNumber;
  token: Token;
  guild: Guild;
  isStakeAmountValid: boolean;
  createTransaction: (
    summary: string,
    txFunction: () => Promise<providers.TransactionResponse>
  ) => void;
}

export interface BalanceInfoLineProps {
  token: Token;
  roundedBalance: number;
}

export interface LockTimeInfoLineProps {
  guild: Guild;
}

export interface VotingPowerInfoLineProps {
  isStakeAmountValid: boolean;
  userVotingPower: BigNumber;
  guild: Guild;
  stakeAmount: BigNumber;
}

export interface UnlockDateInfoLineProps {
  guild: Guild;
  isStakeAmountValid: boolean;
  isRepGuild: boolean;
}
