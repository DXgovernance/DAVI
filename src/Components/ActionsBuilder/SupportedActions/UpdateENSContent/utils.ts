import contenthash from 'content-hash';
import { MAINNET_ID, LOCALHOST_ID, GOERLI_ID } from 'utils';
import { utils } from 'ethers';

const DEFAULT_NAMEHASH =
  '0x0000000000000000000000000000000000000000000000000000000000000000';

export const convertToNameHash = (name: string) => {
  if (!name) {
    return DEFAULT_NAMEHASH;
  }
  return utils.namehash(name);
};

export const convertToContentHash = (ipfsHash: string) => {
  if (!ipfsHash) {
    return '';
  }
  return contenthash.fromIpfs(ipfsHash);
};

export const convertToIpfsHash = (contentHash: string) => {
  if (!contentHash) return null;
  return contenthash.decode(contentHash);
};

export const getIpfsUrl = (ipfsHash: string) => {
  const ipfsRoot = 'ipfs://';
  return `${ipfsRoot}${ipfsHash}`;
};

export const isValidChainId = (chainId: number) => {
  if (!chainId || chainId === LOCALHOST_ID) {
    return MAINNET_ID;
  } else {
    return chainId;
  }
};

export const isAvailableOnENS = (chainId: number) => {
  const validChainId = isValidChainId(chainId);
  const ensNetworks = [MAINNET_ID, GOERLI_ID];
  return ensNetworks.includes(validChainId);
};
