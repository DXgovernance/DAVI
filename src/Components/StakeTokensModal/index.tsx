export { default as StakeTokensModal } from './StakeTokensModal';