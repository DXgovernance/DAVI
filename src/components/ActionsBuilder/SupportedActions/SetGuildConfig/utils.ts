import { FIELDS } from './constants';
import { BigNumber } from 'ethers';
import { SetGuildConfigFields, ControlField } from './types';

export const bn = (n: string | number | BigNumber) => BigNumber.from(n);

type UpdatedValuesReturn = {
  [key in ControlField]: {
    newValue: BigNumber;
    currentValue: BigNumber;
  };
};
export const getUpdatedValues = (
  current: SetGuildConfigFields,
  modifyed: SetGuildConfigFields
): UpdatedValuesReturn | {} => {
  if (!current || !modifyed) return {};
  return Object.keys(current).reduce((acc, key) => {
    if (!FIELDS.map(f => f.name).includes(key)) return acc;
    const currentValue = bn(current[key] || 0);
    const newValue = bn(modifyed[key] || 0);

    if (!currentValue.eq(newValue)) {
      return {
        ...acc,
        [key]: { newValue, currentValue },
      };
    }
    return acc;
  }, {});
};

export const pickUpdatedOrDefaultValue = (
  current: BigNumber,
  modifyed: BigNumber
): BigNumber => {
  if (BigNumber.isBigNumber(current)) {
    if (BigNumber.isBigNumber(modifyed) && !bn(modifyed).eq(current)) {
      return bn(modifyed);
    }
    return bn(current);
  } else if (BigNumber.isBigNumber(modifyed)) {
    return bn(modifyed);
  }
  return bn(0);
};
