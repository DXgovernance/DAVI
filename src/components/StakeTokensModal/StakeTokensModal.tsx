import { Loading } from 'components/primitives/Loading';
import { Modal } from 'components/primitives/Modal';
import { useTranslation } from 'react-i18next';
import { StakeTokensModalProps } from './types';

const StakeTokensModal: React.FC<StakeTokensModalProps> = ({
  isOpen,
  onDismiss,
  token,
  children,
}) => {
  const { t } = useTranslation();
  return (
    <Modal
      header={
        token ? (
          t('stakeTokens', { token: token.symbol })
        ) : (
          <Loading loading text skeletonProps={{ width: '100px' }} />
        )
      }
      isOpen={isOpen}
      onDismiss={onDismiss}
      maxWidth={300}
    >
      {children}
    </Modal>
  );
};

export default StakeTokensModal;
