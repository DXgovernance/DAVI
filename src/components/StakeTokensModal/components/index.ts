export { BalanceWidgetWrapper } from './BalanceWidgetWrapper';
export { StakeTokensForm } from './StakeTokensForm';
export { StakeTokensButton } from './StakeTokensButton';
export {
  LockTimeInfoLine,
  BalanceInfoLine,
  VotingPowerInfoLine,
  UnlockDateInfoLine,
} from './StakeTokensInfoLine';
