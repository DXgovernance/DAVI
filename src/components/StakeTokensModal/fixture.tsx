import { BigNumber } from 'ethers';
import { GuildConfigProps } from 'hooks/Guilds/ether-swr/guild/useGuildConfig';
import { ERC20 } from 'types/contracts';
import { ERC20Guild } from 'types/contracts/ERC20Guild';
import StakeTokensForm from './components/StakeTokensForm/StakeTokensForm';

export const mockStakeTokensFormProps = {
  token: {
    name: 'Mock Token',
    allowance: BigNumber.from(1000000),
    balance: BigNumber.from(1000000),
    info: {
      decimals: 18,
      symbol: 'REP',
      name: 'mockName',
      totalSupply: BigNumber.from(0),
    },
    contract: {
      approve: () => {},
    } as unknown as ERC20,
  },
  userVotingPower: BigNumber.from(0),
  guild: {
    contract: {} as ERC20Guild,
    config: {
      name: 'mockGuildName',
      totalLocked: BigNumber.from(0),
      tokenVault: 'mockTokenVault',
      lockTime: BigNumber.from(0),
    } as unknown as GuildConfigProps,
    totalLocked: BigNumber.from(0),
  },
  createTransaction: () => {},
  isRepGuild: false,
};

export const mockStakeTokensModalProps = {
  isOpen: true,
  onDismiss: () => {},
  StakeTokensForm: StakeTokensForm,
  StakeTokensFormProps: mockStakeTokensFormProps,
};
