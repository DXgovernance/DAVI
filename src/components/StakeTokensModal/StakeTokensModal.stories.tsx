import { ComponentStory, ComponentMeta } from '@storybook/react';
import { StakeTokensModal } from 'components/StakeTokensModal';
import { mockStakeTokensModalProps } from './fixture';

export default {
  title: 'StakeTokensModal',
  component: StakeTokensModal,
  argTypes: {
    isOpen: {
      description: 'State for checking if modal is open',
    },
    onDismiss: {
      description: 'Closing the modal',
    },
    StakeTokensForm: {
      description: 'Form for staking tokens',
    },
    StakeTokensFormProps: {
      description: 'Props for staking tokens form',
    },
  },
} as ComponentMeta<typeof StakeTokensModal>;

const Template: ComponentStory<typeof StakeTokensModal> = args => (
  <StakeTokensModal {...args} />
);

export const Simple = Template.bind({});
Simple.args = mockStakeTokensModalProps;

export const Loading = Template.bind({});
Loading.args = {};
