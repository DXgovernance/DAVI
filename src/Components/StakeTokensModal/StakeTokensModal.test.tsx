import StakeTokensModal from './StakeTokensModal';
import { render } from '../../utils/tests';
import { mockStakeTokensModalProps } from './fixture';

jest.mock('ipfs', () => jest.fn());
jest.mock('cids', () => jest.fn());
jest.mock('orbit-db', () => jest.fn());
jest.mock('ipfs-only-hash', () => jest.fn());

describe('StakeTokensModal', () => {
  it('StakeTokensModal renders properly', () => {
    console.error = jest.fn();
    const { container } = render(
      <StakeTokensModal {...mockStakeTokensModalProps} />,
      {
        container: document.body,
      }
    );
    expect(container).toMatchSnapshot();
  });
});
