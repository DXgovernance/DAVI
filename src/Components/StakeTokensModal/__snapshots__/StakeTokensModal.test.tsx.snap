// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StakeTokensModal StakeTokensModal renders properly 1`] = `
<body>
  <div>
    <div
      class="sc-kEjbQP bzazPo backdropModal"
    >
      <div
        class="sc-jrAFXE llhtMl"
      >
        <div
          class="sc-iqHYmW djTBga"
        >
          
           
          <div
            class="sc-crrszt cBSLGO"
          >
            <h2
              class="sc-jSgvzq sc-bqyKOL cRjjSL jifZfj"
            >
              Stake Mock Token tokens
            </h2>
            <svg
              class="sc-kstqJO gdCTxS"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="18"
                x2="6"
                y1="6"
                y2="18"
              />
              <line
                x1="6"
                x2="18"
                y1="6"
                y2="18"
              />
            </svg>
          </div>
          <div
            class="sc-fodVek fovbhg"
          >
            <div
              class="sc-fHuLxr fXhPFW"
              data-testid="stake-tokens-modal"
            >
              <div
                class="sc-dtTHTy iKlLJk"
              >
                <img
                  alt="DXdao Logo"
                  class="sc-dkIXZx fOVRXm"
                  src="dxdao-icon.svg"
                />
                <h2
                  class="sc-jSgvzq sc-XhUvE cRjjSL gTlifZ"
                >
                  mockGuildName
                </h2>
              </div>
              <div
                class="sc-ikPAEB eIsnaP"
              >
                a few seconds staking period
              </div>
              <div
                class="sc-tYqdw hCwNAv"
              >
                <div
                  class="sc-biBsmb fiOhMx"
                >
                  <span
                    class="sc-eHfRjS ckVAkE"
                  >
                    balance
                    :
                  </span>
                  <span
                    class="sc-hzMMVR TvdeM"
                  >
                    0
                     
                    REP
                  </span>
                </div>
                <div
                  class="sc-biBsmb fiOhMx"
                >
                  <div
                    class="sc-bZSRNg bTohGJ"
                  >
                    <div
                      class="sc-iUuxjF kfXzoC"
                    />
                    <input
                      autocomplete="off"
                      autocorrect="off"
                      class="sc-jcVcSv fxFyvz sc-xyEDr kleKny"
                      data-testid="stake-amount-input"
                      inputmode="decimal"
                      maxlength="79"
                      minlength="1"
                      pattern="^[0-9]*[.,]?[0-9]*$"
                      placeholder="0.0"
                      spellcheck="false"
                      type="text"
                      value="0.0"
                    />
                    <div
                      class="sc-iUuxjF exPtkN"
                    />
                  </div>
                  <button
                    class="sc-dlfnuX dEJhZx"
                    data-testid="stake-amount-max-button"
                  >
                    max
                  </button>
                </div>
              </div>
              <div
                class="sc-biBsmb fiOhMx"
              >
                <span
                  class="sc-eHfRjS ckVAkE"
                >
                  balance
                </span>
                <span
                  class="sc-hzMMVR TvdeM"
                >
                  0
                   
                  REP
                </span>
              </div>
              <div
                class="sc-biBsmb fiOhMx"
              >
                <span
                  class="sc-eHfRjS ckVAkE"
                >
                  yourVotingPower
                </span>
                <span
                  class="sc-hzMMVR TvdeM"
                >
                  -
                </span>
              </div>
              <div
                class="sc-biBsmb fiOhMx"
              >
                <span
                  class="sc-eHfRjS ckVAkE"
                >
                  unlockDate
                </span>
                <span
                  class="sc-hzMMVR TvdeM"
                >
                  -
                </span>
              </div>
              <button
                class="sc-dlfnuX sc-dWddBi dEJhZx coQCtX"
                data-testid="lock-token-spending"
                disabled=""
              >
                Lock
                 
                REP
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
   
</body>
`;
